framework/src/main/java/org/checkerframework/framework/type/poly/AbstractQualifierPolymorphism.java

```diff
@@ -200,7 +200,8 @@ public void resolve(MethodInvocationTree tree, AnnotatedExecutableType type) {
               collector.visit(atypeFactory.getReceiverType(tree), type.getReceiverType()));
     }
 
-    if (instantiationMapping != null && !instantiationMapping.isEmpty()) {
+    if ((instantiationMapping != null && !instantiationMapping.isEmpty())
+        || TreeUtils.isZeroArgumentVarArgs(tree)) {
```

Comment:
Isn't this special case just as relevant for all varargs methods, not just those with no formal parameters before the varargs one?

Reply:
Good catch